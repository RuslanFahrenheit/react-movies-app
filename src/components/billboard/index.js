export { default as Billboard } from './billboard';
