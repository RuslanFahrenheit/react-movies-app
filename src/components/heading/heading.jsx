import React from 'react';
import PropTypes from 'prop-types';
import './heading.scss';

const validHeadingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];

const Heading = ({
  level,
  title,
}) => {
  const safeHeading = level ? level.toLowerCase() : '';
  const HeadingTag = validHeadingLevels.includes(safeHeading) ? safeHeading : 'p';

  return (
    <HeadingTag
      className={`heading heading--${HeadingTag}`}
    >
      {title}
    </HeadingTag>
  );
};

Heading.propTypes = {
  level: PropTypes.oneOf(validHeadingLevels).isRequired,
  title: PropTypes.string.isRequired,
};

export { Heading };
