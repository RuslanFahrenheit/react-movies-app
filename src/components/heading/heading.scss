.heading {
  width: 100%;
  margin: 0;
  font-weight: 200;
  color: #fff;
  text-transform: uppercase;

  &--h2 {
    font-size: 45px;
  }

  &--h6 {
    font-size: 20px;
  }
}
