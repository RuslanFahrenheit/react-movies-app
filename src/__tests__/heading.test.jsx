import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Heading } from '../components/heading';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render a heading', () => {
  act(() => {
    render(
      <Heading
        title="add movie"
        level="h1"
      />,
      container,
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
