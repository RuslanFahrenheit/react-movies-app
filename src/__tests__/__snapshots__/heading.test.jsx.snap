// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a heading 1`] = `"<h1 class=\\"heading heading--h1\\">add movie</h1>"`;
